astropy/visualization/wcsaxes/coordinate_helpers.py

```diff
@@ -186,10 +186,12 @@ def grid(self, draw_grid=True, grid_type=None, **kwargs):
 
         self.grid_lines_kwargs.update(kwargs)
 
-        if self.grid_lines_kwargs["visible"]:
-            if not draw_grid:
-                self.grid_lines_kwargs["visible"] = False
-        else:
+        if draw_grid:
+            self.grid_lines_kwargs["visible"] = True
+        elif draw_grid is False:
+            self.grid_lines_kwargs["visible"] = False
+        elif draw_grid is None:
+            draw_grid = True
```

Comment:
I think the first two lines should be merged, and an `else` clause added, as follows
```suggestion
        if isinstance(draw_grid, bool):
            self.grid_lines_kwargs["visible"] = draw_grid
        elif draw_grid is None:
            draw_grid = True
        else:
            raise TypeError(f"draw_grid expected a boolean or None, got {draw_grid}")
```